### PR TITLE
updates to player filters

### PR DESCRIPTION
**Pull request by dan13ram**
The design/UX has been fully reviewed by Dave. 

Latest deployment here: https://the-game-git-feat-filters-metafam.vercel.app

### CODE REVIEW

File: packages/web/lib/hooks/players.ts [hunk 1]
```diff
@@ -99,14 +64,63 @@ export const usePlayerFilter = (): PlayerFilter => {
...
+      searchFilterUsed ||
+      availabilityFilterUsed ||
+      skillIdsFilterUsed ||
+      timezonesFilterUsed,
+    [
+      playerTypesFilterUsed,
+      searchFilterUsed,
+      availabilityFilterUsed,
+      skillIdsFilterUsed,
+      timezonesFilterUsed,
```
Comment (alalonde) [inline packages/web/lib/hooks/players.ts:120]: Seems like a lot of boilerplate.. I don't have a better solution though :D

**Review by alalonde — APPROVED** [review]
In the preview, when I search for a single skill "front-end development", there is a graphql error. Also, it seems like we probably want more opacity on the filter bar background when scrolling. It looks a bit weird to me

File: packages/web/components/Player/PlayerFilter.tsx [hunk 2]
```diff
@@ -41,182 +119,256 @@ export const PlayerFilter: React.FC<Props> = ({
...
   };
+
+  const { filtersUsed } = useFiltersUsed(queryVariables);
+
+  const [isElementSticky, setIsSticky] = useState<boolean>(false);
+  const ref = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    const cachedRef = ref.current as Element;
+    const observer = new IntersectionObserver(
```
[the remark this answers is not in the document]
Comment (dan13ram) [reply packages/web/components/Player/PlayerFilter.tsx:130]: @alalonde https://caniuse.com/intersectionobserver

**Comment by dan13ram** [comment]
@alalonde I'm able to reproduce the bug locally. After some debugging I have narrowed down the issue.
It has something to do with the schema changed in the return from boxProfile. And its failing only for userId `dekan`.
I've added an issue https://github.com/MetaFam/TheGame/issues/655 to track this.

**Pull request merged**